common/src/main/java/org/corfudb/common/metrics/micrometer/MeterRegistryProvider.java

```diff
@@ -25,6 +26,26 @@ private MeterRegistryProvider() {
      * Class that initializes the Meter Registry.
      */
     public static class MeterRegistryInitializer extends MeterRegistryProvider {
+
+        /**
+         * Create a new instance of MeterRegistry with the given logger, loggingInterval
+         * and clientId.
+         * @param logger A configured logger.
+         * @param loggingInterval A duration between log appends for every metric.
+         * @param clientId An id of a client for this metric.
+         * @return A new meter registry.
+         */
+        public static MeterRegistry newInstance(Logger logger, Duration loggingInterval,
```

Comment:
Can you remove usages to `getInstance` and just instantiate a new instance for LR (just like you do it for the client/server). Having a singleton implemented with a `synchronized` on the getter is not efficient.

Reply:
On the server-side, this would entail passing the registry as a parameter to a lot of classes that require instrumentation. This would make code more difficult to read, this would also require a lot of modifications. I would prefer to leave the registry as a singleton but I need to guarantee that the instance can be configured only once.